source/extract.cpp

```diff
@@ -24,72 +26,67 @@
     }  // namespace
 
     void extract(const std::string& filename, const std::string& workingPath, int overwriteInis, std::function<void()> func)
     {
-        zipper::Unzipper unzipper(filename);
-        std::vector<zipper::ZipEntry> entries;
+        chdir(workingPath.c_str());
+        unzFile zfile = unzOpen(filename.c_str());
+        unz_global_info gi = {0};
+        unzGetGlobalInfo(zfile, &gi);
+
+        ProgressEvent::instance().setTotalSteps(gi.number_entry);
+        ProgressEvent::instance().setStep(0);
+
+        for (int i = 0; i < gi.number_entry; ++i) {
+            char filename_inzip[0x301] = {0};
+            unz_file_info file_info = {0};
+            unzOpenCurrentFile(zfile);
+            unzGetCurrentFileInfo(zfile, &file_info, filename_inzip, sizeof(filename_inzip), NULL, 0, NULL, 0);
+            std::string filename_inzip_s = filename_inzip;
+
+            if (filename_inzip[strlen(filename_inzip) - 1] == '/') {
+                DIR *dir = opendir(filename_inzip);
+            }
 
-        preWork(unzipper, workingPath, entries);
-        std::set<std::string> ignoreList = fs::readLineByLine(FILES_IGNORE);
+            else {
+                FILE *outfile;
+                void *buf = malloc(WRITE_BUFFER_SIZE);
 
-        for (const auto& entry : entries) {
-            if (ProgressEvent::instance().getInterupt()) {
-                break;
-            }
-            if ((overwriteInis == 0 && entry.name.substr(entry.name.length() - 4) == ".ini") || find_if(ignoreList.begin(), ignoreList.end(), [&entry](std::string ignored) {
-                                                            u8 res = ("/" + entry.name).find(ignored);
-                                                            return (res == 0 || res == 1); }) != ignoreList.end()) {
-                if (!std::filesystem::exists("/" + entry.name)) {
-                    unzipper.extractEntry(entry.name);
+                if ((filename_inzip_s == "atmosphere/package3") || (filename_inzip_s == "atmosphere/stratosphere.romfs")) {
+                    outfile = fopen((filename_inzip_s + ".aio").c_str(), "wb");
                 }
-            }
-            else if (entry.name == "atmosphere/stratosphere.romfs" || entry.name == "atmosphere/package3") {
-                std::ofstream readonlyFile(entry.name + ".aio");
-                unzipper.extractEntryToStream(entry.name, readonlyFile);
-            }
-            else {
-                unzipper.extractEntry(entry.name);
-                if (entry.name.substr(0, 13) == "hekate_ctcaer") {
-                    fs::copyFile("/" + entry.name, UPDATE_BIN_PATH);
-                    if (CurrentCfw::running_cfw == CFW::ams && util::showDialogBoxBlocking(fmt::format("menus/utils/set_hekate_reboot_payload"_i18n, UPDATE_BIN_PATH, REBOOT_PAYLOAD_PATH), "menus/common/yes"_i18n, "menus/common/no"_i18n) == 0) {
-                        fs::copyFile(UPDATE_BIN_PATH, REBOOT_PAYLOAD_PATH);
-                    }
```

Comment:
This is a feature, it shouldn't be removed

Reply:
Added